rviz_default_plugins/src/rviz_default_plugins/displays/marker/marker_common.cpp

```diff
@@ -153,9 +154,32 @@ void MarkerCommon::addMessage(const visualization_msgs::msg::Marker::ConstShared
 void MarkerCommon::addMessage(
   const visualization_msgs::msg::MarkerArray::ConstSharedPtr array)
 {
+  using ns_type = decltype(visualization_msgs::msg::Marker::ns);
+  using id_type = decltype(visualization_msgs::msg::Marker::id);
+  using pair_type = std::pair<id_type, const ns_type &>;
+
+  // Keep track of unique markers
+  std::set<pair_type> unique_markers;
+  bool found_duplicate = false;
+
   for (auto const & marker : array->markers) {
+    if (!found_duplicate) {
+      pair_type pair(marker.id, marker.ns);
+      found_duplicate = !unique_markers.insert(pair).second;
+    }
     addMessage(std::make_shared<visualization_msgs::msg::Marker>(marker));
   }
+
+  // Can't use setMarkerStatus on individual markers because processAdd would clear it.
+  const char * kDuplicateStatus = "Duplicate Marker Check";
+  if (found_duplicate) {
+    display_->setStatusStd(
+      rviz_common::properties::StatusProperty::Error,
+      kDuplicateStatus,
+      "Multiple Markers in the same MarkerArray message had the same namespace and id");
```

Comment:
Is this enough information for someone to debug this problem?

Maybe it should include how many are duplicates and/or which namespace/id are offending.

Reply:
I made the error message include the first offending namespace and id in ba16715d08558469e3d2e9e4d8b70f294af710b7